### PR TITLE
add python3-pyotp

### PR DESCRIPTION
**Pull request by lennewbrook**
Please add the following dependency to the rosdep database.

## Package name:

python3-pyotp

## Package Upstream Source:

(https://github.com/pyotp/pyotp)

## Purpose of using this:

This dependency is being used to add a one time password creation package for python.

Distro packaging links:

## Links to Distribution Packages

- Debian: https://packages.debian.org/
  - https://packages.debian.org/bookworm/python3-pyotp
- Ubuntu: https://packages.ubuntu.com/
  - https://packages.ubuntu.com/source/jammy/python-pyotp
- Fedora: https://packages.fedoraproject.org/
  - https://packages.fedoraproject.org/pkgs/python-pyotp/python3-pyotp/fedora-40.html
- Arch: https://www.archlinux.org/packages/
  - https://gitlab.archlinux.org/archlinux/packaging/packages/python-pyotp
- FreeBSD: https://pkg.freebsd.org/
  - https://cgit.freebsd.org/ports/tree/security/py-pyotp
- OpenSUSE: https://software.opensuse.org/
  - https://build.opensuse.org/package/show/openSUSE:Leap:15.6/python3-pyotp

### CODE REVIEW

**Review by mjcarroll — APPROVED** [review]

File: rosdep/python.yaml [hunk 1]
```diff
@@ -8247,6 +8247,13 @@ python3-pyosmium:
   fedora: [python3-osmium]
   nixos: [python3Packages.pyosmium]
   ubuntu: [python3-pyosmium]
+python3-pyotp:
+  arch: [python-pyotp]
+  debian: [python3-pyotp]
+  fedora: [python3-pyotp]
+  freebsd: [py311-pyotp]
+  opensuse: [python3-pyotp]
```
Comment (clalancette) [inline rosdep/python.yaml:8255]:
```suggestion
  rhel:
    '*': [python3-pyotp]
    '8': null
  opensuse: [python3-pyotp]
```

**Review by clalancette — CHANGES_REQUESTED** [review]
This looks like it is available for `RHEL-9`

**Review by clalancette — DISMISSED** [review]
For reasons unrelated to this PR, this needs a rebase, then it should pass CI.

File: rosdep/python.yaml [hunk 2]
```diff
@@ -8303,6 +8303,16 @@ python3-pyosmium:
...
   ubuntu: [python3-pyosmium]
+python3-pyotp:
+  arch: [python-pyotp]
+  debian: [python3-pyotp]
+  fedora: [python3-pyotp]
+  freebsd: [py311-pyotp]
+  rhel:
+    '*': [python3-pyotp]
+    '8': null
+  opensuse: [python3-pyotp]
```
Comment (sloretz) [inline rosdep/python.yaml:8314]: Looks like CI is failing here: 

```
    ERR: list out of alphabetical order line 8314.  'opensuse' should come before 'rhel'
```

**Review by sloretz — DISMISSED** [review]

**Review by clalancette — APPROVED** [review]

**Review by audrow — APPROVED** [review]

**Pull request merged**